### PR TITLE
Badges

### PR DESCRIPTION
**Pull request by PRABgit01**
### Badges on Avtar 
Screenshot 2023-12-22 at 3 14 11 PM

### Badges on icon
Screenshot 2023-12-22 at 3 14 22 PM

### CODE REVIEW

File: src/stories/Badges/Badge.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import { ImageContainer, AvatarImage, BadgeIcon,BadgeForIcon,IconImageBadge } from "./Badge.style";
+
+const Badge = ({ url, type, variant, colorInput,text }) => {
```
Comment (tarakhant1297) [inline src/stories/Badges/Badge.js:4]: @PRABgit01 file name and folder name should be same

File: src/stories/Badges/Badge.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  if (variant === "square") {
+    return (
+      <ImageContainer>
+        <AvatarImage type={type} src={url} alt="avatar" />
+        <BadgeIcon colorInput={colorInput} type={type} />
```
Comment (tarakhant1297) [inline src/stories/Badges/Badge.js:9]: @PRABgit01 give proper name to colorInput

File: src/stories/Badges/Badge.style.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import styled, { css } from "styled-components";
+import theme from "../theme";
+
+const sizes = {
```
Comment (tarakhant1297) [inline src/stories/Badges/Badge.style.js:4]: @PRABgit01 move this to size.js file

**Review by tarakhant1297 — CHANGES_REQUESTED** [review]

File: src/stories/Badges/Badge.styled.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+export const StyledBadgeIcon = styled.div`
+  position: absolute;
+  ${({ type }) => css`
+    top: ${sizes[type]?.top}%;
+    left: ${sizes[type]?.left}%;
+  `}
+  transform: translateX(45%);
+  width: 10px;
+  height: 10px;
+  border: 2px solid;
```
Comment (ankitadrixus08) [inline src/stories/Badges/Badge.styled.js:32]: use shorthand prop @PRABgit01

File: src/stories/Badges/Badge.style.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+export const BadgeIcon = styled.div`
+  position: absolute;
+  ${({ type }) => css`
+    top: ${sizes[type]?.top}%;
+    left: ${sizes[type]?.left}%;
+  `}
+  transform: translateX(45%);
+  width: 10px;
+  height: 10px;
+  border: 2px solid;
```
[the remark this answers is not in the document]
Comment (PRABgit01) [reply src/stories/Badges/Badge.style.js]: i have done the changes

File: src/stories/Badges/Badge.style.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+export const BadgeForIcon = styled(BadgeIcon)`
+  background-color: ${({ colorInput }) => theme?.[colorInput]};
+  ${({ type }) => css`
+    top: ${sizes[type]?.topIcon}%;
+    left: ${sizes[type]?.leftIcon}%;
+  `}
+  width: 15px;
+  height: 15px;
+  border-radius: 50%;
+  color: white;
```
[the remark this answers is not in the document]
Comment (PRABgit01) [reply src/stories/Badges/Badge.style.js]: here i have imported the the color from the theme file

**Review by ankitadrixus08 — APPROVED** [review]

File: src/stories/Badges/sizes.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+const sizes = {
+  small: { height: 2.4, top: 59, left: 37, topIcon: 9, leftIcon: 44 },
```
Comment (tarakhant1297) [inline src/stories/Badges/sizes.js:2]: @PRABgit01 why you add `top: 59, left: 37, topIcon: 9, leftIcon: 44`

**Review by tarakhant1297 — CHANGES_REQUESTED** [review]

File: src/stories/Badges/Badge.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  StyledBadgeForIcon,
+  StyledIconImageBadge,
+} from "./Badge.styled";
+
+const Badge = ({ url, type, variant, colorInput, count }) => {
+  if (variant === "square") {
+    return (
+      <StyledImageContainer>
+        <StyledAvatarImage type={type} src={url} alt="avatar" />
+        <StyledBadgeIcon colorInput={colorInput} type={type} />
```
Comment (tarakhant1297) [inline src/stories/Badges/Badge.js:15]: @PRABgit01 give proper name to `colorInput `

**Review by tarakhant1297 — CHANGES_REQUESTED** [review]

**Review by tarakhant1297 — APPROVED** [review]

**Pull request merged**